datalad/distribution/tests/test_dataset.py

```diff
@@ -323,3 +323,112 @@ def test_Dataset_flyweight(path1, path2):
             ds3 = Dataset('linked')
             ok_(ds3 == ds1)
             ok_(ds3 is not ds1)
+
+
+@with_tempfile
+def test_property_reevaluation(repo1):
+
+    from os.path import lexists
+    from datalad.tests.utils import ok_clean_git
+
+    ds = Dataset(repo1)
+    assert (ds.repo is None)
```

Comment:
pytest leakage?  Shouldn't these asserts (and their unnecessary parentheses :] ) be converted to the nose assert helpers to get more informative errors?